src/lightning_app/cli/lightning_cli.py

```diff
@@ -298,10 +299,17 @@ def _run_app(
                 "Caching is a property of apps running in cloud. "
                 "Using the flag --no-cache in local execution is not supported."
             )
+        if secret:
+            raise click.ClickException(
+                "Secrets can only be used for apps running in cloud. "
+                "Using the flag --secret in local execution is not supported."
+            )
 
     env_vars = _format_input_env_variables(env)
     os.environ.update(env_vars)
 
+    secrets = _format_input_env_variables(secret)
```

Comment:
nit: in case of errors, `_format_input_env_variables()` will raise error messages for the environment variables